src/firmware/app/primitives/chick_motion.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "firmware/app/world/firmware_world.h"
+
+/**
+ * Chick motion figures out how to line the robot up for a kick or chip, but it is up
+ * to users of these helper functions to arm the appropriate autokick and autochip
+ */
+
+/**
+ * <------- B <-------------------------
+ *        ^       ^                 ^         \
+ * alignment   destination   chick motion path \
+ *   angle                                      \
+ *                                               \
+ *                                                A
+ *                                                ^
+ *                                              start
+ */
+
+/**
+ * Start the chick motion in the given world
+ *
+ * @param void_state_ptr [in] A pointer to this primitives state, as allocated by the
+ *                            `create_state` function
+ * @param world [in] The world this primitive is running in.
+ * @param x_destination The x destination for shooting
+ * @param y_destination The y destination for shooting
+ * @param alignment_angle The angle to align to
```

Comment:
What's the consistency on `[in] / [out]`?

Reply:
we'll figure it out in this issue: https://github.com/UBC-Thunderbots/Software/issues/1650